pkg/config/helpers/general.go

```diff
@@ -0,0 +1,58 @@
+package helpers
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+// ResolvePaths updates the given refs to be absolute paths, relative to the given base directory
+func ResolvePaths(refs []*string, base string) error {
+	for _, ref := range refs {
+		// Don't resolve empty paths
+		if len(*ref) > 0 {
+			// Don't resolve absolute paths
+			if !filepath.IsAbs(*ref) {
+				*ref = filepath.Join(base, *ref)
+			}
+		}
+	}
+	return nil
+}
+
+func MakeRelative(path, base string) (string, error) {
+	if len(path) > 0 {
+		rel, err := filepath.Rel(base, path)
+		if err != nil {
+			return path, err
+		}
+		return rel, nil
+	}
+	return path, nil
+}
+
+// RelativizePathWithNoBacksteps updates the given refs to be relative paths, relative to the given base directory as long as they do not require backsteps.
+// Any path requiring a backstep is left as-is as long it is absolute.  Any non-absolute path that can't be relativized produces an error
+func RelativizePathWithNoBacksteps(refs []*string, base string) error {
+	for _, ref := range refs {
+		// Don't relativize empty paths
+		if len(*ref) > 0 {
+			rel, err := MakeRelative(*ref, base)
+			if err != nil {
+				return err
+			}
+
+			// if we have a backstep, don't mess with the path
+			if strings.HasPrefix(rel, "../") {
```

Comment:
will break on windows (if we care or ever plan to use this in CLI)

Reply:
> will break on windows (if we care or ever plan to use this in CLI)

Hasn't come up.  I'm inclined to leave it.  I'm also of the opinion that I only like to support forward slashes and windows users can just deal with it.

Reply:
Can I quote "I'm also of the opinion that I only like to support forward slashes and windows users can just deal with it." ?